tests/v2/validation/auth/openldap_test.go

```diff
@@ -0,0 +1,74 @@
+package auth
+
+import (
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/auth"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type OLDAPTestSuite struct {
+	suite.Suite
+	session *session.Session
+	client  *rancher.Client
+}
+
+func (o *OLDAPTestSuite) TearDownSuite() {
+	o.session.Cleanup()
+}
+
+func (o *OLDAPTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	o.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(o.T(), err)
+
+	o.client = client
+}
+
+func (o *OLDAPTestSuite) TestEnableOLDAP() {
+	subSession := o.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := o.client.WithSession(subSession)
+	require.NoError(o.T(), err)
+
+	a, err := auth.NewAuth(client, subSession)
+	require.NoError(o.T(), err)
+
+	err = a.OLDAP.Enable()
+	require.NoError(o.T(), err)
+
+	ldapConfig, err := client.Management.AuthConfig.ByID("openldap")
+	require.NoError(o.T(), err)
+
+	assert.Truef(o.T(), ldapConfig.Enabled, "Checking if Open LDAP is enabled")
```

Comment:
As discussed offline, can we add these two tests also - annotation and secret assertions ?

Reply:
Added annotation and does secret exist assertions, as an extra added a cross-reference assertion between giving configuration service account password and stored password in the secret.